templates/writing-task-topics.rst

```diff
@@ -0,0 +1,240 @@
+  picked up automatically from the interface of the `__init__` method
+  and will not require developer input.
+  
+.. _task-topics-run:
+	  
+Run method
+----------
+
+- Similarly to the Class initialization, the description of the
+  interface for calling the primary entrypoint function of the class
+  will be picked up automatically from the interface of the `run`
+  method and will not require developer input.
+
+  
+.. _task-topics-debug:
+
+Debugging
+=========
+
+- Also similarly to the Class initialization, information on the
+  debugging parameters, specifically their name, type, and
+  description, will be picked up automatically from docstrings in the
+  class definition.
+
+.. _task-topics-examples:
+   
+Examples
+========
+
+.. Note:: Making this can be a substantial job which requires writing
+          an actual example and then going through and describing line
+          by line in comments inside of it what the example is doing.
+
+
+The example should be very stripped down and use only the basic functionality of the Task.
+
+To give some specific guidance, we will give a few pointers for how
+one might write an example for IsrTask, which we will call
+`exampleIsrTask.py`, then describe it with reST.
+
+In this particular case, we need to use some functions which are
+normally in the `utils.py` class, and to make it more transparent, we
+might want to strip this down and rewrite them locally in the
+`exampleUtils.py` class.
+
+Next, we describe some of the details for the content of `exampleIsrTask.py`.
+
+Where needed, when describing code, the python code block directive can be used as so:
+
+.. code-block:: rst
+
+  .. code-block:: python
+
+    #Create the isr task with modified config
+    isrConfig = IsrTask.ConfigClass()
+    isrConfig.doBias = False #We didn't make a zero frame
+    isrConfig.doDark = True
+    isrConfig.doFlat = True
+    isrConfig.doFringe = False #There is no fringe frame for this example
+		   
+Then, to describe the block setting up configuration parameters, we can say in reST:
+
+.. code-block:: rst
+		
+ The first line: `isrConfig = IsrTask.ConfigClass()` indicates this is
+ a section about setting up the configuration that the code will be run
+ with.  The next several set up specific flags, indicating that we will
+ not do bias or fringing corrections in this code, but will do the dark
+ and flat corrections.
+
+We can then describe the other intermediate sections in ways similar to
+the above, saying that the final output is created with the call to
+the `IsrTask` `run` method:
+
+.. code-block:: python
+		
+       output = isrTask.run(rawExposure, dark=darkExposure, flat=flatExposure)
+
+
+.. _task-topics-algorithm:
+   
+Algorithm details
+====================
+
+This should be written in detailed form and can refer to prior written
+documentation as long as it is accessible by all potential code users.
+Mathematical notation can be used here and written in LaTex through the `math`
```

Comment:
The official spelling appears to be LaTeX actually (https://en.wikipedia.org/wiki/LaTeX) but i'll simplify and go with Latex.

Accepted your changes for referring to 'math' directive.